Branch-SDK/src/main/java/io/branch/referral/network/BranchRemoteInterfaceUrlConnection.java

```diff
@@ -65,7 +66,8 @@ private BranchResponse doRestfulGet(String url, int retryNumber) throws BranchRe
             connection.setConnectTimeout(timeout);
             connection.setReadTimeout(timeout);
 
-            String requestId = getRequestIdFromHeader(connection);
+            String requestId = connection.getHeaderField(Defines.HeaderKey.RequestId.getKey());
+            Branch.getInstance().setCloseRequestNeeded(Boolean.parseBoolean(connection.getHeaderField(Defines.HeaderKey.SendCloseRequest.getKey())));
```

Comment:
What happens if the header doesn't exist? Or if the field isn't parseable as a Boolean?

Reply:
if header doesn't exist we get a null, if we try to parse the null, we get false, which is the default :)